bitarray/_bitarray.c

```diff
@@ -2050,17 +2051,17 @@ bitarray_cpinvert(bitarrayobject *self)
     return res;
 }
 
-#if IS_GCC
-#define BITWISE_FUNC_INTERNAL(SELF, OTHER, OP, OPEQ) do {             \
-    Py_ssize_t i = 0;                                                 \
-                                                                      \
-    for (; i + V16C_SIZE < Py_SIZE(SELF); i += V16C_SIZE) {           \
-        simd_v16uc_op((SELF)->ob_item + i, (OTHER)->ob_item + i, OP); \
-    }                                                                 \
-                                                                      \
-    for (; i < Py_SIZE(SELF); ++i) {                                  \
-        (SELF)->ob_item[i] OPEQ (OTHER)->ob_item[i];                  \
-    }                                                                 \
+#if HAS_VECTORS
+#define BITWISE_FUNC_INTERNAL(SELF, OTHER, OP, OPEQ) do {         \
+    Py_ssize_t i = 0;                                             \
+                                                                  \
+    for (; i + sizeof(vec) < Py_SIZE(SELF); i += sizeof(vec)) {   \
+        vector_op((SELF)->ob_item + i, (OTHER)->ob_item + i, OP); \
+    }                                                             \
+                                                                  \
```

Comment:
Clever way to use the same counter for bulk operations and then the trailing bits.